src/qt/peertablemodel.cpp

```diff
@@ -54,24 +54,21 @@ class PeerTablePriv
     void refreshPeers()
     {
         {
-            TRY_LOCK(cs_vNodes, lockNodes);
-            if (!lockNodes)
-            {
-                // skip the refresh if we can't immediately get the lock
-                return;
-            }
             cachedNodeStats.clear();
+            std::vector<CNodeStats> vstats;
+            if(g_connman)
+                g_connman->GetNodeStats(vstats);
 #if QT_VERSION >= 0x040700
-            cachedNodeStats.reserve(vNodes.size());
+            cachedNodeStats.reserve(vstats.size());
 #endif
-            Q_FOREACH (CNode* pnode, vNodes)
+            Q_FOREACH (const CNodeStats& nodestats, vstats)
             {
                 CNodeCombinedStats stats;
                 stats.nodeStateStats.nMisbehavior = 0;
                 stats.nodeStateStats.nSyncHeight = -1;
                 stats.nodeStateStats.nCommonHeight = -1;
                 stats.fNodeStateStatsAvailable = false;
-                pnode->copyStats(stats.nodeStats);
+                stats.nodeStats = nodestats;
```

Comment:
For better modularity and readability, I presume. Instead of iterating through all `vNode`s holding a lock, a method has been introduced to get copy of stats. This can potentially allow to completely encapsulate how stats are stored and access them only through method.

Although this code is less efficient,
* amount of data copied is small,
* code is not on critical path.